### PR TITLE
Add possibility to get the amount of jobs with a certain uuid or tag …

### PR DESCRIPTION
**Pull request by hons82**
…in all queues

This allows to check is a certain job is running, or waiting to be scheduled

References: #264

### CODE REVIEW

File: Sources/SwiftQueue/SqOperationQueue.swift [hunk 1]
```diff
@@ -136,6 +136,14 @@ internal final class SqOperationQueue: OperationQueue {
             operation.cancel()
         }
     }
+    
+    func operationCount(tag: String) -> Int {
+        return operations.compactMap({ $0 as? SqOperation }).filter({ $0.info.tags.contains(tag) }).count
+    }
+    
+    func operationCount(uuid: String) -> Int {
```
Comment (lucas34) [inline Sources/SwiftQueue/SqOperationQueue.swift]: Job ID is supposed to be unique right ? So I don't get the point of returning a Int. Shouldn't be boolean ?

File: Sources/SwiftQueue/SqOperationQueue.swift [hunk 2]
```diff
@@ -136,6 +136,14 @@ internal final class SqOperationQueue: OperationQueue {
...
+        return operations.compactMap({ $0 as? SqOperation }).filter({ $0.info.uuid == uuid }).count
```
Comment (lucas34) [inline Sources/SwiftQueue/SqOperationQueue.swift]: To check if a job exist I would rather do it in a fully swift way than with High order function. Should be more optimised

```swift
    public func containsJob(uuid: String) -> Bool {
        for element: OperationQueue in manage.values {
            for case let item as SqOperation in element.operations where item.info.uuid == uuid {
                return true
            }
        }
        return false
    }
```

File: Sources/SwiftQueue/SqOperationQueue.swift (hunk 1 above)
Comment (hons82) [reply Sources/SwiftQueue/SqOperationQueue.swift]: I did not check all the code, but is that uniqueness enforced? 
or is it just supposed to be unique?
Nevertheless it should be bool!

File: Sources/SwiftQueue/SqOperationQueue.swift (hunk 2 above)
Comment (hons82) [reply Sources/SwiftQueue/SqOperationQueue.swift]: I'll update that

File: Sources/SwiftQueue/SqOperationQueue.swift (hunk 1 above)
Comment (lucas34) [reply Sources/SwiftQueue/SqOperationQueue.swift]: There is a unique UUID constraint per queue to enforce this

Comment (hons82) [reply Sources/SwiftQueue/SqOperationQueue.swift]: I see...
Ok. I've updated the PR

File: Sources/SwiftQueue/SwiftQueueManager.swift [hunk 3]
```diff
@@ -105,6 +105,26 @@ public final class SwiftQueueManager {
         }
     }
 
+    /// number of jobs with this tag in all queues
+    public func jobCount(tag: String) -> Int {
+        assertNotEmptyString(tag)
+        var count = 0
+        for element in manage.values {
+            count += element.operationCount(tag: tag)
+        }
```
Comment (lucas34) [inline Sources/SwiftQueue/SwiftQueueManager.swift]: This part could also be written as 

```swift
for case let item as SqOperation in element.operations where item.info.tags.contains(tag) {
    count += 1
}
```

Then we don't need to change the code of SqOperation

**Comment by lucas34** [comment]
Please also add test cases for the new added code

**Comment by lucas34** [comment]
It's now possible to do it by using `getAll()` method.

**Comment by hons82** [comment]
My solution to this has been an extension to check if the uuid is returned by getAll

```
import Foundation
import SwiftQueue

public extension SwiftQueueManager {
    /// check if there is jobs with this uuid in at least one queues
    /// - Parameter uuid: the uuid to check for
    func containsJob(uuid: String) -> Bool {
        for element in getAll().values {
            for case let info in element where info.uuid == uuid {
                return true
            }
        }
        return false
    }
}

```

This is however not possible in v5.0.0 because Jobinfo does not contain a uuid anymore. If I understood it correct, the uuid is now in the constraints, but those are not accessible.

**Comment by lucas34** [comment]
@hons82  is it because getConstraint is internal ?
https://github.com/lucas34/SwiftQueue/blob/e0e80c86d17bc280dcb0d152d2bfeeffcd34e8f8/Sources/SwiftQueue/Utils.swift#L145

I think it should be public

**Pull request closed without merging**